go/vt/vtgate/planbuilder/operators/horizon_planning.go

```diff
@@ -128,19 +118,25 @@ func optimizeHorizonPlanning(ctx *plancontext.PlanningContext, root ops.Operator
 		case *Horizon:
 			return pushOrExpandHorizon(ctx, in)
 		case *Projection:
-			return tryPushingDownProjection(ctx, in)
+			return tryPushProjection(ctx, in)
 		case *Limit:
-			return tryPushingDownLimit(in)
+			return tryPushLimit(in)
 		case *Ordering:
-			return tryPushingDownOrdering(ctx, in)
+			return tryPushOrdering(ctx, in)
 		case *Aggregator:
-			return tryPushingDownAggregator(ctx, in)
+			return tryPushAggregator(ctx, in)
 		case *Filter:
-			return tryPushingDownFilter(ctx, in)
+			return tryPushFilter(ctx, in)
 		case *Distinct:
-			return tryPushingDownDistinct(in)
+			return tryPushDistinct(in)
 		case *Union:
-			return tryPushDownUnion(ctx, in)
+			return tryPushUnion(ctx, in)
+		case *SubQueryContainer:
+			return pushOrMergeSubQueryContainer(ctx, in)
+		case *QueryGraph:
+			return optimizeQueryGraph(ctx, in)
```

Comment:
Why is ``*QueryGraph`` being used here? It seems that the ``*QueryGraph`` type has already been handled in the previous transformToPhysical method.

Reply:
Great question! This is because we might have subqueries hiding inside horizons, and these won't become available until we do horizon expansion.

I'm not really happy with how this ended up, and I think I want to clean up this part. We shouldn't produce `Horizon`s that are hiding subqueries - these should get expanded straight away. Unfortunately, we depend on the root being a `Horizon` to know what columns the user originally asked for in `addTruncationOrProjectionToReturnOutput`. I'll work on cleaning this up in coming PRs